### PR TITLE
Support inference for GCU

### PR DESCRIPTION
**Pull request by EnflameGCU**
配置`GCU`设备的推理流程

### CODE REVIEW

File: tools/infer/utility.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 import random
 from ppocr.utils.logging import get_logger
 
+import paddle_custom_device.gcu.passes as gcu_passes
```
Comment (GreatV) [inline tools/infer/utility.py:28]: Regular users don't really need `paddle_custom_device`.

Comment (EnflameGCU) [reply tools/infer/utility.py]: You are right, this needs to be corrected. Thank you for the correction.

File: tools/infer/utility.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ def init_args():
     parser.add_argument("--use_xpu", type=str2bool, default=False)
     parser.add_argument("--use_npu", type=str2bool, default=False)
     parser.add_argument("--use_mlu", type=str2bool, default=False)
+    parser.add_argument("--use_gcu", type=str2bool, default=False)
```
Comment (jzhang533) [inline tools/infer/utility.py:44]: please include a short message to help user understand its purpose.
To me, at the first glance, I am guessing gcu stands for GraphCore Unit, realized it's Enflame's device after searching. 

Other arguments(i.e.: uie_xpu, use_npu, use_mlu) also need help message to clarify the respective devices, but could be addressed in separate PRs.

Comment (EnflameGCU) [reply tools/infer/utility.py]: OK, thanks for pointing that out. I'll add comments to GCU first. 

I am afraid that our description of other arguments may not be accurate, and they may need to be described by the developers of the corresponding hardware manufacturers.

File: tools/infer/utility.py [hunk 3]
```diff
@@ -41,6 +41,9 @@ def init_args():
     parser.add_argument("--use_xpu", type=str2bool, default=False)
     parser.add_argument("--use_npu", type=str2bool, default=False)
     parser.add_argument("--use_mlu", type=str2bool, default=False)
+    parser.add_argument(
+        "--use_gcu", type=str2bool, default=False
+    )  # Use Enflame GCU(General Compute Unit)
```
Comment (jzhang533) [inline tools/infer/utility.py:46]:
```suggestion
    parser.add_argument(
        "--use_gcu", type=str2bool, default=False,
        help="Use Enflame GCU(General Compute Unit)",
    )
```

File: tools/infer/utility.py (hunk 2 above)
Comment (jzhang533) [reply tools/infer/utility.py]: > I am afraid that our description of other arguments may not be accurate, and they may need to be described by the developers of the corresponding hardware manufacturers.

agree

File: tools/infer/utility.py (hunk 3 above)
Comment (EnflameGCU) [reply tools/infer/utility.py]: Thank you for your suggestion.

File: tools/infer/utility.py [hunk 4]
```diff
@@ -293,6 +299,28 @@ def create_predictor(args, mode, logger):
             config.enable_custom_device("mlu")
         elif args.use_xpu:
             config.enable_xpu(10 * 1024 * 1024)
+        elif args.use_gcu:  # for Enflame GCU(General Compute Unit)
+            import paddle_custom_device.gcu.passes as gcu_passes
```
Comment (cuicheng01) [inline tools/infer/utility.py:303]: Should we first check whether the `paddle_custom_device` package is present? If it’s not, remind the user to install it first?

File: tools/infer/utility.py [hunk 5]
```diff
@@ -293,6 +299,28 @@ def create_predictor(args, mode, logger):
...
+                    "gcu", 0, paddle.inference.PrecisionType.Half
+                )
+                gcu_passes.set_exp_enable_mixed_precision_ops(config)
+            else:
+                config.enable_custom_device("gcu")
+
+            if paddle.framework.use_pir_api():
+                config.enable_new_ir(True)
+                config.enable_new_executor(True)
+                kPirGcuPasses = gcu_passes.inference_passes(
```
Comment (nepeplwu) [inline tools/infer/utility.py:317]: Why do we need to build passes based on "PaddleOCR" or "PaddleXXX"?

File: tools/infer/utility.py (hunk 4 above)
Comment (EnflameGCU) [reply tools/infer/utility.py:303]: Thank you for your suggestion.

File: tools/infer/utility.py (hunk 5 above)
Comment (EnflameGCU) [reply tools/infer/utility.py:317]: We need to optimize performance based on hardware features and even network structure, and the processing here is to increase this flexibility.

**Review by cuicheng01 — APPROVED** [review]

**Pull request merged**